x-pack/plugins/fleet/dev_docs/developing_kibana_and_fleet_server.md

```diff
@@ -0,0 +1,400 @@
+# Developing Kibana and Fleet Server simulatanously
+
+Many times, a contributor to Fleet will only need to make changes to [Fleet Server](https://github.com/elastic/fleet-server) or [Kibana](https://github.com/elastic/kibana) - not both. But, there are times when end-to-end changes across both componenents are necessary. To facilitate this, we've created a guide to help you get up and running with a local development environment that includes both Kibana and Fleet Server. This is a more involved process than setting up either component on its own.
+
+This guide seeks to get you up and running with the following stack components for local development:
+
+- Kibana (from source)
+- Elasticsearch (from a snapshot)
+- Fleet Server (from source, in standalone mode)
+
+Getting this development environment up and running will allow you to make changes to both Kibana and Fleet Server simultaneously to develop and test features end-to-end.
+
+Before continuing, please review the developer documentation for both Fleet Server and Kibana. Make sure your local workstation can run each service independently. This means setting up Go, Node.js, Yarn, resolving dependencies, etc.
+
+- https://github.com/elastic/fleet-server?tab=readme-ov-file#development
+- https://github.com/elastic/kibana/blob/main/CONTRIBUTING.md
+
+You can either run your local stack over HTTPS or HTTP. Each has its own set of instructions below. An HTTPS environment more closely resembles production, but requires a bit more setup. HTTPS may be necessary when developing certain features related to certificates. An HTTP environment is easier to set up, and will generally be the fastest and easiest way to get up and running for end-to-end development.
+
+_This guide expects you have forked + cloned the https://github.com/elastic/fleet-server and https://github.com/elastic/kibana repoisitories locally._
+
+## Running a local stack over HTTP
+
+1. In your `kibana` directory, run a local Elasticsearch server from the latest snapshot build by running:
+
+```bash
+# Set a data path to prevent blowing away your Elasticsearch data between server restarts
+$ yarn es snapshot --license trial -E path.data=/tmp/es-data
+```
+
+2. Add the following to your `kibana.dev.yml`
+
+```yaml
+server.basePath: '/some-base-path' # Optional, if used, sets basePath in kibana url e.g. https://localhost:5601/some-base-path/app/fleet
+server.versioned.versionResolution: oldest
+elasticsearch.hosts: [http://localhost:9200]
```

Comment:
is this the same setting as `xpack.fleet.agents.elasticsearch.hosts`? AFAIK it is not needed since there is a default es output in line 71

Reply:
This is the global setting for the ES host Kibana should communicate with. I think this might be the default value but it seems harmless to include.